src/libprojectM/ProjectM.cpp

```diff
@@ -586,16 +543,16 @@ int ProjectM::initPresetTools(int gx, int gy)
 
     std::string url = (m_flags & FLAG_DISABLE_PLAYLIST_LOAD) ? std::string() : settings().presetURL;
 
-    if ((m_presetLoader = new PresetLoader(gx, gy, url)) == 0)
+    if ((m_presetLoader = std::make_unique<PresetLoader>(gx, gy, url)) == 0)
```

Comment:
On failure to allocate, `make_unique` (and `new` too) will throw `std::bad_alloc`, not return `nullptr`: https://en.cppreference.com/w/cpp/memory/unique_ptr/make_unique

Reply:
Correct. I think I've replaced the logic in most, but not all places. Also comparing with `0` instead of using `nullptr` is not a great thing.